### DISCUSSION TITLE
When using JAX, how to print out the underlying XLA logs?

### DISCUSSION THREAD

**Discussion by knightXun** [jax-ml/jax#21477] [Q&A] [answered]
I want to know what things the XLA exactly do

**Comment by hawkinsp** [comment, marked as answer]
Well, I'd start with `XLA_FLAGS=--xla_dump_to=/somewhere`, which will dump before and after-optimization HLO and assembly. Does that help?

**Reply by knightXun** [reply]
You helps me a lot！ Thanks!
